### PR TITLE
feat: ui now connects to backend, full e2e

### PR DESCRIPTION
**Pull request by sjaanus**
Added API hooks and now frontend actually sends data to database.

### CODE REVIEW

File: frontend/src/component/feedbackNew/FeedbackContext.ts [hunk 1]
```diff
@@ -2,33 +2,25 @@ import { createContext } from 'react';
...
+    IFeedbackText;
 
-export const FeedbackContext = createContext<IFeedbackContext>({
-    feedbackData: DEFAULT_FEEDBACK_DATA,
-    showFeedback: true,
-    setShowFeedback: setShowFeedback,
-    openFeedback: openFeedback,
-    closeFeedback: closeFeedback,
-});
+export const FeedbackContext = createContext<IFeedbackContext | undefined>(
```
Comment (sjaanus) [inline frontend/src/component/feedbackNew/FeedbackContext.ts:24]: Refactored context, we do not need default fake data here.

File: frontend/src/component/feedbackNew/FeedbackComponent.tsx [hunk 2]
```diff
@@ -1,7 +1,12 @@
-import { Box, Button, styled, TextField } from '@mui/material';
+import { Box, Button, IconButton, styled, TextField, Tooltip } from '@mui/material';
 import { ConditionallyRender } from 'component/common/ConditionallyRender/ConditionallyRender';
 import { useFeedback } from './useFeedback';
-import React from 'react';
+import React, { useState } from 'react';
+import CloseIcon from '@mui/icons-material/Close';
+import produce from 'immer';
```
Comment (kwasniew) [inline frontend/src/component/feedbackNew/FeedbackComponent.tsx:6]: are you sure you want to add immer for such a simple use case?

Comment (sjaanus) [reply frontend/src/component/feedbackNew/FeedbackComponent.tsx]: Removed

File: frontend/src/component/feedbackNew/FeedbackComponent.tsx [hunk 3]
```diff
@@ -131,33 +139,78 @@ const StyledScoreValue = styled('label')(({ theme }) => ({
...
+    const { setToastData } = useToast();
+    const { addFeedback } = useUserFeedbackApi();
+
+
+
+    const onSubmission = async (event: React.FormEvent<HTMLFormElement>) => {
+        event.preventDefault();
+        const formData = new FormData(event.currentTarget);
+        const data = Object.fromEntries(formData);
+        await addFeedback(data as any);
```
Comment (kwasniew) [inline frontend/src/component/feedbackNew/FeedbackComponent.tsx:167]: In the example I sent before the author was using zod type to parse data. We can try with the orval generated type. Can it parse data?

**Review by kwasniew — APPROVED** [review]

**Pull request merged**